contracts/core/EntryPoint.sol

```diff
@@ -375,6 +401,31 @@ contract EntryPoint is IEntryPoint, StakeManager {
     }
     }
 
+    /**
+     * revert if either account deadline or paymaster deadline is expired
+     */
+    function _validateDeadline(uint opIndex, uint256 deadline, address paymaster, uint256 paymasterDeadline) internal view {
+        //we want to treat "zero" is "maxint", so we subtract one, ignoring overflow
+    unchecked {
+        // solhint-disable-next-line not-rely-on-time
+        if ((deadline - 1) < block.timestamp) {
```

Comment:
we handle specific value (zero) this way. its not a general case... should we remove it?

Reply:
I think it's fine, just a bit unusual.  The comment explains it but I assume it'll come up.

Reply:
```
if (deadline == SIG_VALIDATION_FAILED){
 revert();
} else if (deadline == 0) {
  // no-op
} else if (deadline < block.timestamp){
  revert();
}
```

Would this not work? Is it in any way worse then `(deadline - 1) < block.timestamp)`? 

This logic repeats twice - for Account and for Paymaster. Won't it be wise to extract the repeating part to a utility method?

Reply:
separate if's are possible (it was actually that way to begin with). but indeed, 0 is treated as "infinite" value. its so hard to pass such value in solidity, that I let the users put "0" instead..

the logic is repeated twice, on different fields. I did want to merge them to one (we care only about min(deadline, paymasterDeadline), after treating zero as infinite..), but since either can be SIG separately, I couldn't, and the logic to "merge" these values has to be off-chain.